pkg/idp/service.go

```diff
@@ -143,6 +143,12 @@ func (s *Service) CreateResource(ctx context.Context, data *Configuration) ([]*C
 	}
 
 	var idp *Configuration
+
+	// assign random ID if empty
+	if data.ID == "" {
+		data.ID = uuid.NewString()
+	}
```

Comment:
issue: when creating new entitites, ID should never be user specified since it may conflict with backend-generated one (especially in the future). In case a user specified an ID, we should reject with bad request
Same goes for the other change